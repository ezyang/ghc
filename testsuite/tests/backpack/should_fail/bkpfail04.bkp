unit p where
    signature A where
        data A = A { foo :: Int }

unit q where
    signature A where
        data A = A { bar :: Bool }

unit r where
    include p
    include q
    module M where
        import A
        x = foo
        y = bar
