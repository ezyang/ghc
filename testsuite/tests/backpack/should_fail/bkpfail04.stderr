[1 of 3] Processing p
/--- Shape for p
provides:
requires: A -> p(hole:A):A
               A{A, A, foo}
signature p(hole:A):A
\---
  [1 of 1] Compiling A[abstract sig]  ( p/A.hsig, nothing )
[2 of 3] Processing q
/--- Shape for q
provides:
requires: A -> q(hole:A):A
               A{A, A, bar}
signature q(hole:A):A
\---
  [1 of 1] Compiling A[abstract sig]  ( q/A.hsig, nothing )
[3 of 3] Processing r
/--- Shape for r
provides: M -> r(hole:A):M
               x, y
requires: A -> p(hole:A):A, q(hole:A):A
               A{A, foo, A, bar}
include p(hole:A) [] requires [(A, [p(hole:A):A])]
include q(hole:A) [] requires [(A, [q(hole:A):A])]
module r(hole:A):M
\---
  [1 of 3] Including p
  [2 of 3] Including q

bkpfail04.bkp:11:5: error:
    The record label lists for ‘A’ differ
    data A = A {foo :: GHC.Types.Int}
    data A = A {bar :: GHC.Types.Bool}
      Promotable
